### ISSUE TITLE
Unable to install updates

### ISSUE THREAD

**Issue by eposse** [gitahead/gitahead#222]
Problem description
-------------------

Unable to install GitAhead updates.

Version
-------

GitAhead 2.5.5

Environment
-----------

macOS Mojave 10.14.5

Steps to reproduce
------------------

1. Launch GitAhead
2. "Update GitAhead" window appears showing new updates available. Click on [Install Update]
3. After downloading the update, click [Install and Restart]

Expected result
---------------

The update is downloaded and installed.

Actual result
-------------

The main window disappears and a dialog pops up with the error message "Unable to install update: The existing bundle could not be moved to the trash"

The window disappears, although GitAhead is still running. The update is not installed.

**Comment by hackhaslam** [comment]
Are you running GitAhead as the same user that owns the .app bundle?

**Comment by eposse** [comment]
Yes. I tried reinstalling 2.5.5 again and this is what happens:

1. I open the GitAhead-2.5.5.dmg
2. A dialog pops up with two icons, one for GitAhead, one for the Applications folder. I drag GitAhead to the Applications folder.
3. It asks me to Authenticate, as the Applications folder requires admin access. I provide the admin credentials and GitAhead.app is added to Applications as expected.
4. I start GitAhead. The first time it asks me to confirm. I select [Open].

Then the updates dialog pops-up and the behaviour is the same as above: unable to install the update.

By the way, since I tried this a few times, each time it downloaded the update, where does it stores these updates?

Thanks

**Comment by hackhaslam** [comment]
Is your user an administrator (e.g. in System Preferences -> Users & Groups)? If you're an administrator then you shouldn't need to enter your password to add and remove from /Applications. Unfortunately, GitAhead's update mechanism doesn't know how to request elevated privileges. I can look at this at some point but it probably won't happen soon.

**Comment by eposse** [comment]
I see. My user is a standard user, not admin. Perhaps the installer should allow for local, non-admin, non-system wide installations.

So the only workaround for now is to manually remove GitAhead to install a new version?

If so, where can I find the updates downloaded to completely remove traces of the old version?

Thanks

**Comment by hackhaslam** [comment]
Yes, GitAhead should have a way to request elevated privileges so that it can install the update, but it doesn't have that now.

The only workaround is to manually remove GitAhead. The app bundle is self-contained. Just move it to the trash to uninstall.

**Comment by pbrunnen** [comment]
@hackhaslam 
If I could suggest, it might be useful to have a link to manually download the update presented in the update notice dialog (maybe on the version number).  At least as a convenience until the permissions issue is resolved.

Thanks! -Peter.
